docs/design/free_credit_offer_typesDesignDoc.md

```diff
@@ -0,0 +1,253 @@
+    award_credit_duration_days - int
+    Invitee_credit_duration_days - int
+    type - enum[FREE_TIER, REFERRAL]
+    status - enum[ON_GOING, DEFAULT, EXPIRED, NO_STATUS]
+    PRIMARY KEY (id)
+```
+
+**user_credit table**
+```sql
+    User_id - bytea
+    offer_id - int
+    credits_earned - float
+    Credit_type - enum[AWARD, INVITEE, NO_TYPE]
+    expires_at - timestamp
+    Created_at - timestamp
+    referred_by - bytea (nullable)
+    FOREIGN KEY (offer_id)
+    PRIMARY KEY (user_id)
+```
+
+**user table**
+```sql
+    total_Referred - int
+```
+
+### Offer Program Service
+
+**satellite/offer/offer.go**
+- Create offers interface to interact with offer table
+
+```golang
+type Offers interface {
+  ListAllOffers(ctx context.Context) ([]Offer, error)
+  GetOfferById(ctx context.Context, offerId []byte) (Offer, error)
+  Update(ctx context.Context, offer *Offer) (*Offer, error)
+  Delete(ctx context.Context, offerId []byte)
+  Create(ctx context.Context, offer *Offer)
+}
+```
+
+**satellite/console/credit.go**
+- Create a user_credit interface to interact with the user_credit table
+
+```golang
+type Credits interface {
+  AvailableCredits(ctx context.Context, userId uuid.UUID) (int, error)
+  ListByCreditType(ctx context.Context, userId uuid.UUID, creditType Credit.Type) ([]Credit, error)
+  Update(ctx context.Context, credit *Credit) (*Credit, error)
+  Create(ctx context.Context, credit *Credit) (*Credit, error)
+}
+```
+
+**satellite/console/database.go**
+- New method for retrieving user credit data from user_credit table
+  
+```golang
+type DB interface {
+	// Users is a getter for Users repository
+  Users() Users
+  // Credits is a getter for Credits repository
+  Credits() Credits
+	// Projects is a getter for Projects repository
+	Projects() Projects
+	// ProjectMembers is a getter for ProjectMembers repository
+	ProjectMembers() ProjectMembers
+	// APIKeys is a getter for APIKeys repository
+	APIKeys() APIKeys
+	// BucketUsage is a getter for accounting.BucketUsage repository
+	BucketUsage() accounting.BucketUsage
+	// RegistrationTokens is a getter for RegistrationTokens repository
+	RegistrationTokens() RegistrationTokens
+	// UsageRollups is a getter for UsageRollups repository
+	UsageRollups() UsageRollups
+
+	// BeginTransaction is a method for opening transaction
+  BeginTx(ctx context.Context) (DBTx, error)
+}
+```
+
+**satellite/offer/offerweb/server.go**
+- Open a new private port on the satellite for admin users to manage referral offer configuration
+
+```golang
+// NewServer creates a new instance of offerweb server
+Func NewServer(logger *zap.Logger, config Config, service *offer.Service, listener net.Listener) *Server {}
+
+// Register offerweb server onto satellite
+peer.Offer.Endpoint = offerweb.NewServer(logger, config, service, listener)
+```
+
+### Referral Links
+
+The referral link url will be a static url that contains userid as the unique identifier
+Exp: `https://mars.tardigrade.io/?uuid=<userid>`
```

Comment:
Is there a reason we are not using subpath here? Adding things at the top level will get annoying fast. Maybe `https://mars.tardigrade.io/r/userid` or `https://mars.tardigrade.io/referral/userid` or something like that?

Reply:
I will change it to `https://mars.tardigrade.io/register/?userid` since we will be using the existing UI for the sign up page